### PR TITLE
ci: add docs.ts test builds

### LINKED ISSUE #3802: Test that docs.rs can build our crates
### Motivation

We found that docs for the `wasmer 3.2.0` crate weren't able to build in docs.rs because of some bugs around conditional compliation. The underlying issue was fixed in #3791, but it'd be nice to make sure CI warns us when this happens. 

### Proposed solution

We can use the `docs.rs` CLI to build things locally ([instructions](https://github.com/rust-lang/docs.rs/blob/master/README.md#build-subcommand)).

In theory, it's "just" a case of running the following:

```console
$ cargo install --git 
$ for $crate in $workspace; do docs-rs  build crate --local /path/to/$crate; done
```

(this should be automatable using the `cargo-metadata` crate and a `cargo xtask` helper)

### Alternatives

We can always leave things as they were done in #3791 and make sure we keep that `Makefile` command in sync with our `[package.metadata.docs.rs]` metadata.

### PR DESCRIPTION
**Pull request by meskill**
For only the packages with specific set of features for docs.rs building

# Description

/claim #3802

### CODE REVIEW

**Comment by syrusakbary** [comment]
It seems tests are failing

**Comment by meskill** [comment]
> It seems tests are failing

fixed now.

[Here](https://github.com/wasmerio/wasmer/actions/runs/4794431337/jobs/8527810140?pr=3819) is the example of failed job that I reproduced by bringing back mistaken code from [fix](https://github.com/wasmerio/wasmer/pull/3791/files)

File: Makefile [hunk 1]
```diff
@@ -440,8 +440,16 @@ endif
...
+test-build-docs-rs:
+	@manifest_docs_rs_features_path="package.metadata.docs.rs.features"; \
+	for manifest_path in lib/*/Cargo.toml; do \
+		toml get "$$manifest_path" "$$manifest_docs_rs_features_path" >/dev/null 2>&1; \
+		if [ $$? -ne 0 ]; then \
+			continue; \
+		fi; \
+		features=$$(toml get "$$manifest_path" "$$manifest_docs_rs_features_path" | sed 's/\[//; s/\]//; s/"\([^"]*\)"/\1/g'); \
+		$(CARGO_BINARY) doc $(CARGO_TARGET_FLAG) --manifest-path "$$manifest_path" --features "$$features" || exit 1; \
+	done
```
[the remark this answers is not in the document]
Comment (meskill) [reply Makefile]: sure, I've added some logging for this

**Comment by meskill** [comment]
Some tests are failing, but I don't think it is related to my changes

**Comment by syrusakbary** [comment]
By looking at the logs, it seems only "lib/compiler" and "lib/api" are being tested. Not sure why, but that needs to be fixed

**Comment by meskill** [comment]
> By looking at the logs, it seems only "lib/compiler" and "lib/api" are being tested. Not sure why, but that needs to be fixed

It is intentional as I wanted to test only packages with `[package.metadata.docs.rs]` features i.e. settings that can affect building documentation on docs.rs. That the issue was about. Also, there is already a job `build-docs` in makefile that will build all documentation with default set of features, although only on master.

**Comment by syrusakbary** [comment]
Got it. Looks good!

**Review by syrusakbary — APPROVED** [review]

**Pull request merged**